### PR TITLE
Update go dependencies

### PR DESCRIPTION
**Pull request by cblecker**
- Update osdk to 0.18.2
- Update client-go and related kubernetes dependencies to 0.19.7
- Remove dependency on k8s.io/kubernetes

### CODE REVIEW

**Comment by jharrington22** [comment]
@cblecker Nice! :bow: 

I'll take a look at the test tomorrow.

**Comment by jharrington22** [comment]
@cblecker figured out the test issues. 

The sharedvolume test delete's the PersistentVolume to test the `uneditSharedVolume` corner cases (immutable PVs since Dedicated Admin's shouldn't be able to delete them). When it does this the sharedvolume controller see's the PV delete event, queues a sharedvolume reconcile which validates there is no PV, requeues another sharedvolume reconcile with no error which then `Ensures` the PV is replaced.

This causes the cached version of the PV and the server version of the PV in the test to become out of sync causing the `Operation cannot be fulfilled on persistentvolumes "pv-namespace-y-sv-b": object was modified` error. 

We need to add the following to fix the test:

```
diff --git a/pkg/controller/sharedvolume/sharedvolume_controller_test.go b/pkg/controller/sharedvolume/sharedvolume_controller_test.go
index fcaf63f..fccdb08 100644
--- a/pkg/controller/sharedvolume/sharedvolume_controller_test.go
+++ b/pkg/controller/sharedvolume/sharedvolume_controller_test.go
@@ -347,6 +347,11 @@ func TestReconcile(t *testing.T) {
        if res, err = r.Reconcile(req); res != test.NullResult || err != nil {
                t.Fatalf("Expected no requeue, no error; got\nresult: %v\nerr: %v", res, err)
        }
+
+       _, pvMap, _ = validateResources(t, r.client, 2)
+
+       pv = pvMap[pvname]
+
        // validateResources proves the PV and PVC came back.
        svMap, pvMap, _ = validateResources(t, r.client, 2)
 
```

**Comment by jharrington22** [comment]
Just noticed the statics test is also failing.

**Comment by jharrington22** [comment]
@cblecker because of the logger changes we to remove the ZapLogger refs.

```
diff --git a/pkg/controller/statics/statics_controller_test.go b/pkg/controller/statics/statics_controller_test.go
index 023d91c..d5036d1 100644
--- a/pkg/controller/statics/statics_controller_test.go
+++ b/pkg/controller/statics/statics_controller_test.go
@@ -27,7 +27,6 @@ import (
 // TODO: Test add()/watches somehow?
 
 func setup() (logr.Logger, *ReconcileStatics) {
-       logf.SetLogger(logf.ZapLogger(true))
 
        // OpenShift types need to be registered explicitly
        scheme.Scheme.AddKnownTypes(securityv1.SchemeGroupVersion, &securityv1.SecurityContextConstraints{})
diff --git a/pkg/controller/statics/statics_test.go b/pkg/controller/statics/statics_test.go
index 83dcf1d..465329a 100644
--- a/pkg/controller/statics/statics_test.go
+++ b/pkg/controller/statics/statics_test.go
@@ -24,7 +24,6 @@ func TestEnsureStatics(t *testing.T) {
        // Future-proof this test against new statics being added.
        checkNumStatics(t)
 
-       logf.SetLogger(logf.ZapLogger(true))
        logger := logf.Log.Logger
        ctx := context.TODO()
```

**Comment by jharrington22** [comment]
:ok_hand: 

```
rator/version
?       openshift/aws-efs-operator/cmd/manager  [no test files]
?       openshift/aws-efs-operator/config       [no test files]
?       openshift/aws-efs-operator/pkg/apis     [no test files]
?       openshift/aws-efs-operator/pkg/apis/awsefs      [no test files]
?       openshift/aws-efs-operator/pkg/apis/awsefs/v1alpha1     [no test files]
?       openshift/aws-efs-operator/pkg/controller       [no test files]
ok      openshift/aws-efs-operator/pkg/controller/sharedvolume  (cached)
ok      openshift/aws-efs-operator/pkg/controller/statics       (cached)
?       openshift/aws-efs-operator/pkg/fixtures [no test files]
?       openshift/aws-efs-operator/pkg/test     [no test files]
ok      openshift/aws-efs-operator/pkg/util     (cached)
?       openshift/aws-efs-operator/version      [no test files]
```

**Comment by jharrington22** [comment]
Nice, I'll now test on cluster.

**Review by jharrington22 — APPROVED** [review]
/lgtm
